### PR TITLE
ci: Add GitHub Actions worflows and remove travis

### PR DESCRIPTION
**Pull request by cpanato**
## Description
ci: Add GitHub Actions worflows and remove travis

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
Part of https://github.com/openfaas/faas/issues/1585

## How Has This Been Tested?
n/a

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by cpanato** [comment]
build ran on my account: https://github.com/cpanato/nats-queue-worker/runs/2289345158?check_suite_focus=true

**Comment by alexellis** [comment]
Hi @cpanato, thank you for this PR.

Could you run through a basic test on your repo please?

Given the way the GH actions work, it templates your username and uploads an image to your own custom GHCR.io repo for testing.

Alex

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+      - name: Push containers
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./Dockerfile
+          outputs: "type=registry,push=true"
+          platforms: linux/amd64,linux/arm/v7,linux/arm64
+          tags: |
+            ghcr.io/openfaas/nats-queue-worker:${{ github.sha }}
```
Comment (alexellis) [inline .github/workflows/publish.yaml:52]: `ghcr.io/openfaas/` <- this should be templated. I'm not sure which repo you copied from, but have a look at another, and you should see how we do it.

I suspect it's my fault for pointing you at the wrong example.. sorry about that.

https://github.com/openfaas/faas/blob/master/.github/workflows/publish.yml

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  push:
+    branches: [ '*' ]
+  pull_request:
+    branches: [ '*' ]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        go-version: [1.13.x]
```
Comment (alexellis) [inline .github/workflows/build.yaml:13]: Please set to 1.15.x

**Comment by cpanato** [comment]
@alexellis the two workflows ran on my account

build: https://github.com/cpanato/nats-queue-worker/runs/2311878376?check_suite_focus=true
publish: https://github.com/cpanato/nats-queue-worker/runs/2311894847?check_suite_focus=true

PTAL when you have some free time

**Review by alexellis — APPROVED** [review]
LGTM

**Comment by alexellis** [comment]
We can also move to `secrets.GITHUB_TOKEN` now instead of individual secrets that I create and manage.

**Pull request merged**